### PR TITLE
feat(e2e): add podchaos api and test cases

### LINKED ISSUE #682: [feature] add support for chaos mesh.
**Is your feature request related to a problem? Please describe.**

We need to take the initiative to introduce some faults(chaos) to test the high availability of the cluster.

**Describe the solution you'd like**

I have introduced the API of Chaos Mesh: https://github.com/radondb/radondb-mysql-kubernetes/pull/681

TODO:
- [ ] podchaos api & test cases  @ChengBaitai
- [ ] netchaos api & test cases  @runkecheng 

official repo:
https://github.com/chaos-mesh/chaos-mesh

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by chengbaitai**
### What type of PR is this?

/enhancement

### Which issue(s) this PR fixes?

Fixes #682 

### What this PR does?

Summary:
Chaos Mesh is a cloud native Chaos Engineering platform that orchestrates chaos on Kubernetes [`Chaos documents.`](https://chaos-mesh.org/zh/docs/).Chaos mesh use CRD defnine many types of fault and provide detailed of parameter description.So wei can mainly focus on life cycle of CRD instead of focus on basic implements of specific faults.
The details of this implements as follows:

- introduce chaos mesh api into e2e
- programming API according to definition of faults
- programming test case

apis:
- [x] podkill
- [x] containerkill
- [x] podfailure

### CODE REVIEW

**Comment by runkecheng** [comment]
ci检查未通过，提交代码前执行：
```
go fmt ./...
go vet ./...
```

**Comment by runkecheng** [comment]
import 规范参考 https://github.com/radondb/radondb-mysql-kubernetes/pull/693#discussion_r970212681

File: test/e2e/chaos/podChaos.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	})
+
+	It("kill leader", func() {
+		killLeader(f)
+		time.Sleep(5 * time.Second)
+		f.WaitClusterReadiness(clusterKey)
+	})
+})
+
+func killLeader(f *framework.Framework) {
```
Comment (runkecheng) [inline test/e2e/chaos/podChaos.go:38]: 其他故障也需要补充

**Comment by runkecheng** [comment]
建议补充成功注入后的资源信息，例如mysql节点，podchaos。

**Review by runkecheng — CHANGES_REQUESTED** [review]

File: test/e2e/framework/chaos.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	podChaos := newPodChaos(option.Name, f.kubectlOptions.Namespace)
+	podChaos.Spec.Action = chaosapis.PodKillAction
+	podChaos.Spec.Mode = option.Mode
+	podChaos.Spec.Selector.LabelSelectors = option.Labels
+
+	Expect(f.Client.Create(context.TODO(), podChaos, &client.CreateOptions{})).Should(Succeed())
+	f.waitChaosInjected(podChaos.Name, 30)
+}
+
+func (f *Framework) KillContainers(option *PodChaosOptions) {
```
Comment (runkecheng) [inline test/e2e/framework/chaos.go:53]: 缺少containers参数输入
```
Spec.containerNames: Invalid value: []string(nil): the name of container is required on container-kill action: invalid value
```

File: test/e2e/framework/chaos.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	podChaos := newPodChaos(option.Name, f.kubectlOptions.Namespace)
+	podChaos.Spec.Action = chaosapis.PodFailureAction
+	podChaos.Spec.Mode = option.Mode
+	podChaos.Spec.Selector.LabelSelectors = option.Labels
+
+	Expect(f.Client.Create(context.TODO(), podChaos, &client.CreateOptions{})).Should(Succeed())
+	f.waitChaosInjected(podChaos.Name, 30)
+}
+
+func (f *Framework) waitChaosInjected(chaosName string, timeout time.Duration) {
```
Comment (runkecheng) [inline test/e2e/framework/chaos.go:92]: 判断注入的过程应该有日志输出

File: test/e2e/framework/chaos.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
+package framework
```
Comment (runkecheng) [inline test/e2e/framework/chaos.go:1]: 补充清理测试环境（测试时创建的chaos）的api

**Review by acekingke — APPROVED** [review]
LGTM

**Review by zhl003 — APPROVED** [review]
LGTM

**Pull request merged**